### DISCUSSION TITLE
Render Markdown to static html rather than SSG

### DISCUSSION THREAD

**Discussion by iMarv** [vercel/next.js#31029] [Help] [answered]
Hey guys,

I am building a small website which includes a blog that gets updated very rarely.
I decided to use markdown for the blogposts and render it with [MDXRemote](https://github.com/hashicorp/next-mdx-remote)

The posts are being rendern through a `[slug].tsx`, which gets metadata and content from the Markdown files through `getStaticProps`, fetching all files with `getStaticPaths`.

This works like a charm, but I do not get over the point that every time I want to view a blogpost, a json with static data is being loaded. All my content is available at build time, there are no external fetches whatsoever. Is there a way to force this file/path to be rendered statically to html in my build step?

I have so far been aware of `next export`, which I would like to give a pass for now as I make heavy use of `next/image` and would prefer to keep the loader for this as it is (except you have an easy solution that I can throw in of course ;) ).

I will attach the file that renders the markdown components, not really sure if it is of use here though.

```ts
const postsDirectory = join(process.cwd(), "posts");

const components = {};

export default function Post({
  source,
  frontMatter,
}: InferGetStaticPropsType<typeof getStaticProps>) {
  return (
    <PostWrapper
      description={frontMatter.description}
      articleImage={frontMatter.imageUrl}
      title={frontMatter.title}
    >
      <MDXRemote {...source} components={components} />
    </PostWrapper>
  );
}

export const getStaticProps: GetStaticProps = async ({ params }) => {
  const postFilePath = join(postsDirectory, `${params?.slug}.mdx`);
  const source = await fs.readFile(postFilePath);

  const { content, data } = matter(source);

  const mdxSource = await serialize(content, {
    // Optionally pass remark/rehype plugins
    mdxOptions: {
      remarkPlugins: [],
      rehypePlugins: [],
    },
    scope: data,
  });

  return {
    props: {
      source: mdxSource,
      frontMatter: data,
    },
  };
};

export const getStaticPaths = async () => {
  const paths = syncFs
    .readdirSync(postsDirectory)
    // Only include md(x) files
    .filter((path) => /\.mdx?$/.test(path))
    // Remove file extensions for page paths
    .map((path) => path.replace(/\.mdx?$/, ""))
    // Map the path into the static paths object required by Next.js
    .map((slug) => ({ params: { slug } }));

  return {
    paths,
    fallback: false,
  };
};
```

`PostWrapper` is a basic component that applies styles and inserts an absolute image path to `/public` coming from the markdown frontmatter.

**Comment by sfiquet** [comment, marked as answer]
Not sure if you still need the answer to this, but here's a short explanation:

If a json file is being downloaded, it's because you use client-side navigation (through the Link component). The Next router does the fetch for you. 

The html file is also generated at build time and is used when the page is requested directly from the browser, i.e. when you reload the page or use a bookmark or just type the url in the address bar. 

The json file is normally much smaller than the html file so it's a smaller download. If you don't want the benefits of client-side navigation, just remove the Link component and use a normal `<a>` link.

**Reply by iMarv** [reply]
My mind is blown, thank you
